File: NYTimes/app/src/main/java/com/example/nytimes/views/SettingActivity.java
package com.example.nytimes.views;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.example.nytimes.R;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class SettingActivity extends AppCompatActivity {

private TextView email;
private FirebaseAuth firebaseAuth;
private TextView logout;
private FirebaseUser user;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);

        email=findViewById(R.id.tvDisplayEmail);
        logout=findViewById(R.id.tvLogOut);
        firebaseAuth=FirebaseAuth.getInstance();
        user=firebaseAuth.getCurrentUser();
        email.setText(user.getEmail());
        logout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                firebaseAuth.signOut();
                startActivity(new Intent(SettingActivity.this,LoginActivity.class));
                finish();
            }
        });

    }
}

File: NYTimes/app/src/main/java/com/example/nytimes/views/SettingActivity.kt
package com.example.nytimes.views

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.example.nytimes.R
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_setting.*

class SettingActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_setting)

        val firebaseAuth = FirebaseAuth.getInstance()
        val user = firebaseAuth.getCurrentUser()
        tvDisplayEmail.setText(user!!.getEmail())

        tvLogOut.setOnClickListener(View.OnClickListener {
            firebaseAuth.signOut()
            startActivity(Intent(this@SettingActivity, LoginActivity::class.java))
            finish()
        })
    }
}